clients/src/main/java/org/apache/kafka/common/quota/QuotaFilter.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.common.quota;
+
+import java.util.Objects;
+
+/**
+ * Describes a quota entity filter.
+ */
+public class QuotaFilter {
+
+    // Matches all entities with the entity type specified.
+    private static final String MATCH_SPECIFIED = "";
+
+    private final String entityType;
+    private final String match;
```

Comment:
Can we use `Optional<String>` for this?  Using magic string values feels messy, and will leak into the API

Reply:
Done.